### ISSUE TITLE
I had written 4 Class for Fiber All Blocking Java-API

### ISSUE THREAD

**Issue by linkerlin** [puniverse/quasar#215]
Could I commit them as a PR?

**Comment by linkerlin** [comment]
PR #216 
Please have a look.

**Comment by fhntv24** [comment]
Is it really async? It's just thread created with task ... Not so much async-ish

**Comment by linkerlin** [comment]
@fhntv24 It's really async.
Every request will be wrappered in a Callable, and Callables will be executed in a thread pool.
No new thread created when a task is submit.

**Comment by fhntv24** [comment]
It's not real async, it's just blocking thread pool - and this what even you sad. You can create real async DNS lookup using NIO, so solution same as JDBC, but I don't think what something better than that is required for DNS lookups

**Comment by linkerlin** [comment]
@fhntv24 UDP is very different with TCP. NIO is useless when socket is a udp socket.

**Comment by fhntv24** [comment]
Yea, useless ... Right ... So what Netty did is magic ... And also DNS never uses TCP, never!

Actually, DNS can use TCP for large request that are larger than one UDP packet, and you can use NIO for UDP. If you want to, you can use NIO even for IP protocol ( but there is no implementation of such, this is just "theory" ). I did use it for UDP, and it's really easy to do. Please, do not say things that you don't know.

If it's soo different, then how DatagramSocket.receive may know what you are the one who did DNS request? Answer is simple, UDP uses ports too (Incoming and outcoming), and even if there is no real connection - dns must answer to you some-how. Connection-less doesn't mean what you cannot use NIO with it.

**Comment by linkerlin** [comment]
@fhntv24 for your infomation : I had released a project BlackholePy as  a DNS proxy which convert UDP DNS requset to TCP。

**Comment by pron** [comment]
Hi. I believe your functionality is already covered by `FiberAsync.runBlocking`. Isn't it?

**Comment by fhntv24** [comment]
@pron yes it is, and @linkerlin just made an bycicle. And he doesn't get what you could write async NIO powered UDP client/server. Also, checkout netty resolver. Yes, netty is pretty big, but maybe it could be an entire library that makes quarsar and netty work together? That would be pretty cool, as there is a lot of netty http clients and other pretty usefull async tools.

**Comment by pron** [comment]
@fhntv24 Please keep your tone respectful; everyone here is trying to help.

**Comment by fhntv24** [comment]
@pron sorry, my bad.

**Comment by linkerlin** [comment]
@pron Happy coding! Waiting for your good news!
